test/end-to-end/core.sh

```diff
@@ -42,6 +42,10 @@ function wait_for_app() {
   wait_for_command '[[ "$(curl -s http://${FRONTEND_IP}:5432/keys/foo)" = "1337" ]]'
 }
 
+# service dns entry is visible via master service
+MASTER_IP="$(dig @${API_HOST} "kubernetes.default.svc.cluster.local." +short A | head -n 1)"
+[ "$(dig @${MASTER_IP} "kubernetes.default.svc.cluster.local." A)" ]
```

Comment:
does this return a non-zero return code on failure or do we need to grep for expected output?